src/provider.rs

```diff
@@ -81,6 +84,22 @@ pub struct Provider {
     pub oauth2_authorizer: Option<Oauth2Authorizer>,
 }
 
+/// Get resolver to query MX records.
+///
+/// We first try resolver_from_system_conf() which reads the system's resolver from `/etc/resolv.conf`.
+/// This does not work at least on some Androids, therefore we use use ResolverConfig::default()
+/// which default eg. to google's 8.8.8.8 or 8.8.4.4 as a fallback.
+async fn get_resolver() -> Result<AsyncStdResolver, ResolveError> {
```

Comment:
It's ok to simply import `anyhow::Result` at the top of the module rather than the test and return it here, so `anyhow::Result` is used as much as possible, preferrably across the whole crate.

Reply:
i also first tried that, getting "expected struct `anyhow::Error`, found struct `ResolveError`" - so i assumed i have to convert the object somehow and did not want to dive into that yesterday :)

Reply:
Something like
```
let resolver = ...?;
Ok(result)
```

`?` will convert the `Result` automatically.

Reply:
that's clever! i did not thought too much about these conversions in the past :)

i did a successor pr that uses map_err() at https://github.com/deltachat/deltachat-core-rust/pull/2853 - but you would prefer the one above?